hyp3_autorift/process.py

```diff
@@ -191,32 +226,28 @@ def process(reference: str, secondary: str, parameter_file: str = DEFAULT_PARAME
         lat_limits, lon_limits = geometry.bounding_box(f'{reference}.zip', polarization=polarization, orbits=orbits)
 
     elif platform == 'S2':
-        gdal.SetConfigOption('GDAL_DISABLE_READDIR_ON_OPEN', 'EMPTY_DIR')
-        gdal.SetConfigOption('AWS_REQUEST_PAYER', 'requester')
-        gdal.SetConfigOption('AWS_REGION', 'eu-central-1')
-        # Also set for new CXX threads in Geogrid/autoRIFT
-        os.environ['GDAL_DISABLE_READDIR_ON_OPEN'] = 'EMPTY_DIR'
-        os.environ['AWS_REQUEST_PAYER'] = 'requester'
-        os.environ['AWS_REGION'] = 'eu-central-1'
-
         reference_metadata = get_s2_metadata(reference)
-        reference_path = reference_metadata['assets']['B08']['href'].replace('s3://', '/vsis3/')
+        reference_path = get_s2_path(reference_metadata)
```

Comment:
`get_s2_path` only uses the single `metadata['assets']['B08']['href']` attribute, the function would be more expressive and easier to test/use if it took `s3_url: str` instead of `metadata: dict`